### ISSUE TITLE
build fails to start if a name of a source secret contains a dot

### ISSUE THREAD

**Issue by mbalazs** [openshift/origin#15876]
When creating a secret from the web console, it is possible to create a secret with dots in its name (e.g. "hello.world"). However, the build controller won't be able to start the builder pod in this case:

```
1m         5m          8         hello-1   Build                 Warning   FailedCreate       {build-controller }   Error creating: Pod "hello-1-build" is invalid: [spec.volumes[2].name: Invalid value: "my.secret-source": must match the regex [a-z0-9]([-a-z0-9]*[a-z0-9])? (e.g. 'my-name' or '123-abc'), spec.containers[0].volumeMounts[2].name: Not found: "my.secret-source"]
```

##### Version
oc v1.5.1
kubernetes v1.5.2+43a9be4
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://api.podspace.io:8443
openshift v1.5.1
kubernetes v1.5.2+43a9be4

##### Steps To Reproduce
1. Click "Add to Project" on the web console to deploy an S2I application (eg. NodeJS)
2. Use the sample GIT repository: https://github.com/openshift/nodejs-ex.git as the source repo URL
3. Create a GIT secret on the deployment form: under "Advanced Options", click on "Create New Secret", and create a basicAuth secret. Enter "my.secret" under Secret Name.
4. Deploy the application

##### Current Result
The application template gets deployed, but the creation of the builder pod fails because of this error: Error creating: Pod "hello-1-build" is invalid: [spec.volumes[2].name: Invalid value: "my.secret-source": must match the regex [a-z0-9]([-a-z0-9]*[a-z0-9])? (e.g. 'my-name' or '123-abc'), spec.containers[0].volumeMounts[2].name: Not found: "my.secret-source"]

##### Expected Result
The build controller should create a working pod template even when the secret name includes a dot, or the web console should not allow a secret to be created which has a dot in its name.

**Comment by mbalazs** [comment]
Yes:

```
[root@master1 ~]# oc export secret my.secret
apiVersion: v1
data:
  password: ....
  username: ....
kind: Secret
metadata:
  creationTimestamp: null
  name: my.secret
type: kubernetes.io/basic-auth
```

This is 100% reproducible by following the steps to reproduce above.

**Comment by bparees** [comment]
sounds like we need to replace dots with dashes when we create the mount name for the secret.

**Comment by jim-minter** [comment]
@mbalazs thanks for the report.  Should be fixed in #15946 when it merges.
